assets/tests/hooks/useDetour.test.ts

```diff
@@ -0,0 +1,132 @@
+import { beforeEach, describe, expect, jest, test } from "@jest/globals"
+import { fetchDetourDirections } from "../../src/api"
+import { renderHook, waitFor } from "@testing-library/react"
+import { useDetour } from "../../src/hooks/useDetour"
+import { act } from "react-dom/test-utils"
+
+jest.mock("../../src/api")
+
+beforeEach(() => {
+  jest.mocked(fetchDetourDirections).mockResolvedValue(null)
+})
+
+describe("useDetour", () => {
+  test("when `addConnectionPoint` is first called, `startPoint` is set", () => {
+    const start = { lat: 0, lon: 0 }
+
+    const { result } = renderHook(useDetour)
+
+    act(() => result.current.addConnectionPoint(start))
+
+    expect(result.current.startPoint).toBe(start)
+  })
+
+  test("when `addConnectionPoint` is called a second time, `endPoint` is set", () => {
+    const start = { lat: 0, lon: 0 }
+    const end = { lat: 1, lon: 1 }
+
+    const { result } = renderHook(useDetour)
+
+    act(() => result.current.addConnectionPoint(start))
+    act(() => result.current.addConnectionPoint(end))
+
+    expect(result.current.startPoint).toBe(start)
+    expect(result.current.endPoint).toBe(end)
+  })
+
+  test("when `startPoint` is null, `addWaypoint` does nothing", () => {
+    const { result } = renderHook(useDetour)
+
+    expect(result.current.startPoint).toBeNull()
+
+    act(() => result.current.addWaypoint({ lat: 0, lon: 0 }))
+
+    expect(result.current.waypoints).toEqual([])
+  })
+
+  test("when `endPoint` is set, `addWaypoint` does nothing", () => {
+    const start = { lat: 0, lon: 0 }
+    const end = { lat: 1, lon: 1 }
+
+    const { result } = renderHook(useDetour)
+
+    expect(result.current.startPoint).toBeNull()
+
+    act(() => result.current.addConnectionPoint(start))
+    act(() => result.current.addConnectionPoint(end))
+
+    act(() => result.current.addWaypoint({ lat: 0, lon: 0 }))
+
+    expect(result.current.waypoints).toEqual([])
+  })
+
+  test("when `addWaypoint` is called, `detourShape` is updated", async () => {
+    const start = { lat: 0, lon: 0 }
+    const end = { lat: 1, lon: 1 }
+
+    jest
+      .mocked(fetchDetourDirections)
+      .mockImplementation((coordinates) =>
+        Promise.resolve({ coordinates: coordinates })
+      )
+
+    const { result } = renderHook(useDetour)
+
+    act(() => result.current.addConnectionPoint(start))
+    act(() => result.current.addWaypoint(end))
+
+    expect(result.current.startPoint).toBe(start)
+
+    expect(jest.mocked(fetchDetourDirections)).toHaveBeenNthCalledWith(1, [
+      start,
+      end,
+    ])
+
+    await waitFor(() =>
+      expect(result.current.detourShape).toStrictEqual([start, end])
+    )
+  })
+
+  test("when `undoLastWaypoint` is called, removes the last `waypoint`", async () => {
```

Comment:
Do we also want a test asserting that when `undoLastWaypoint` is called, the API is called again?

Reply:
9c76ca2af8f7810282a09c3577d9f1b8ced44dcb